### PR TITLE
Added Bitwise Operators

### LINKED ISSUE #75: Adding Bitwise Operators
**Suggestion-**
Bitwise operators should be added to the list for learning Python because they provide essential tools for working with binary data, optimizing code, and performing low-level operations. Understanding bitwise operators is crucial for tasks like setting and clearing specific bits, checking the parity of numbers, and working with hardware interfaces.  

Bitwise operators are a fundamental part of Python programming, especially when it comes to tasks that involve low-level data manipulation, optimization, and binary data. They operate at the individual bit level, making them essential for various programming scenarios. Here are a few reasons why learning bitwise operators in Python is important:  

Binary Data Manipulation: In many applications, such as working with file formats, network protocols, and embedded systems, data is represented in binary form. Bitwise operators allow you to manipulate and extract specific bits or groups of bits within binary data efficiently.  

Optimization: Bitwise operations can be used to optimize code for speed and memory efficiency. For example, using bitwise shifts instead of multiplication/division by powers of 2 can result in faster code execution.

Flags and Masks: Bitwise operators are commonly used to work with flags and masks. Flags are binary values that indicate specific conditions, and masks are used to isolate or modify specific bits within a value. Bitwise operations make it easy to work with these constructs.

### PR DESCRIPTION
**Pull request by ArcXeon**
*Please read the [contributing guidelines](https://github.com/huangsam/ultimate-python/blob/master/CONTRIBUTING.md) before submitting a pull request.*

---

**Describe the change**
Added new file bitwise.py which explains use of bitwise operators in Python.

This commit fixes #75

### CODE REVIEW

**Comment by ArcXeon** [comment]
@huangsam Please review my PR.

File: ultimatepython/syntax/bitwise.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    result_or = a | b  # Binary: 0111 (Decimal: 7)
+    assert result_or == 7
+
+    # Bitwise XOR (^) operator compares each bit of two integers and returns 1
+    # if the bits are different (one is 1 and the other is 0), otherwise returns 0.
+    result_xor = a ^ b  # Binary: 0110 (Decimal: 6)
+    assert result_xor == 6
+
+    # Bitwise NOT (~) operator inverts all bits of an integer. It returns the
+    # one's complement of the integer.
```
Comment (huangsam) [inline ultimatepython/syntax/bitwise.py:28]: Can you please provide a link about one's / two's complement? Something like this:

https://www.geeksforgeeks.org/difference-between-1s-complement-representation-and-2s-complement-representation-technique/

I suspect not everyone is aware of CS concepts like this when first starting out in their programming career.

File: ultimatepython/syntax/bitwise.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+"""
+
+
+def main():
+    # Define some integer values for demonstration
+    a = 5   # Binary: 0101
+    b = 3   # Binary: 0011
+
+    # Bitwise AND (&) operator compares each bit of two integers and returns 1
+    # if both bits are 1, otherwise returns 0.
```
Comment (huangsam) [inline ultimatepython/syntax/bitwise.py:13]: Nitpick: Can we remove the trailing `.` characters for each of the comments in this module?

File: README.md [hunk 3]
```diff
@@ -83,6 +83,7 @@ There are two ways of running the modules:
     - Conditional: [if | if-else | if-elif-else](ultimatepython/syntax/conditional.py) (:cake:)
     - Loop: [for-loop | while-loop](ultimatepython/syntax/loop.py) (:cake:)
     - Function: [def | lambda](ultimatepython/syntax/function.py) (:cake:)
+    - Bitwise: [Bitwise operators](ultimatepython/syntax/bitwise.py) (:cake:)
```
[the remark this answers is not in the document]
Comment (huangsam) [reply README.md:86]: Also should we add this module after the `Expression` lesson?

**Review by huangsam — COMMENTED** [review]
Good start! I provided an initial round of feedback

**Comment by huangsam** [comment]
@ArcXeon let me know if you have any questions or concerns

File: ultimatepython/syntax/bitwise.py (hunk 2 above)
Comment (ArcXeon) [reply ultimatepython/syntax/bitwise.py:13]: Ok will do that.

File: ultimatepython/syntax/bitwise.py (hunk 1 above)
Comment (ArcXeon) [reply ultimatepython/syntax/bitwise.py:28]: Yeah, I think that will be helpful. I will add it.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ArcXeon) [reply README.md:86]: Yeah, I think that will be better.

**Comment by ArcXeon** [comment]
@huangsam I have done the changes.

**Comment by huangsam** [comment]
Excellent great work!

**Pull request merged**